apps/backoffice-v2/src/common/hooks/useHomeLogic/useHomeLogic.ts

```diff
@@ -0,0 +1,24 @@
+import { ComponentProps } from 'react';
+import { DateRangePicker } from '@/common/components/molecules/DateRangePicker/DateRangePicker';
+import { useZodSearchParams } from '@/common/hooks/useZodSearchParams/useZodSearchParams';
+import { HomeSearchSchema } from '@/pages/Home/home-search-schema';
+
+export const useHomeLogic = () => {
+  const [searchParams, setSearchParams] = useZodSearchParams(HomeSearchSchema);
+
+  const handleDateRangeChange: ComponentProps<typeof DateRangePicker>['onChange'] = (range: {
+    start: { toISOString: () => string };
+    end: { toISOString: () => string };
+  }) => {
```

Comment:
ComponentProps<typeof DateRangePicker>['onChange'] gives you the type for `range`. If the type is wrong then we're doing something incorrectly.
```suggestion
  const handleDateRangeChange: ComponentProps<typeof DateRangePicker>['onChange'] = (range) => {
```